### PR TITLE
services are updated during every each sync

### PR DESCRIPTION
**Pull request by DanielZhangQD**
### What problem does this PR solve? 
Fix https://github.com/pingcap/tidb-operator/pull/2284#issuecomment-619485792
And also the `last-applied-configuration` annotation is not set correctly during update for headless services.
### What is changed and how does it work?
Set the `last-applied-configuration` annotation correctly for services.
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
   - `kubectl get svc -A --watch` confirms that services are not updated without any change to service spec.
   - check the log of tidb-controller-manager that no service update log without any change to service spec.

Code changes

 - Has Go code change

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
Fix the issue that services are updated during each sync even if no changes to the service configuration
```

### CODE REVIEW

File: pkg/manager/member/tiflash_member_manager.go [hunk 1]
```diff
@@ -146,7 +146,7 @@ func (tfmm *tiflashMemberManager) syncHeadlessService(tc *v1alpha1.TidbCluster)
 	if !equal {
 		svc := *oldSvc
 		svc.Spec = newSvc.Spec
-		err = controller.SetServiceLastAppliedConfigAnnotation(newSvc)
+		err = controller.SetServiceLastAppliedConfigAnnotation(&svc)
```
Comment (Yisaer) [inline pkg/manager/member/tiflash_member_manager.go:149]: Do we need to set clusterIP here?

Comment (DanielZhangQD) [reply pkg/manager/member/tiflash_member_manager.go:149]: We support setting clusterIP in the `service` of tc, so only need to set the clusterIP if it's not set in the `service` of the tc.

**Comment by DanielZhangQD** [comment]
> I think There is no difference in logic for pd/tikv/tidb/tiflash to sync the service except the svc spec while I see the different logic in each controller. Can we extract the common logic and call the same function to handle this logic?

There are slight differences between the logics now, we can refactor them later.

File: pkg/manager/member/pd_member_manager.go [hunk 2]
```diff
@@ -138,11 +138,13 @@ func (pmm *pdMemberManager) syncPDServiceForTidbCluster(tc *v1alpha1.TidbCluster
 		svc := *oldSvc
 		svc.Spec = newSvc.Spec
 		// TODO add unit test
-		svc.Spec.ClusterIP = oldSvc.Spec.ClusterIP
 		err = controller.SetServiceLastAppliedConfigAnnotation(&svc)
 		if err != nil {
 			return err
 		}
+		if newSvc.Spec.ClusterIP == "" {
```
Comment (weekface) [inline pkg/manager/member/pd_member_manager.go:145]: I think it makes no sense for users to update `clusterIP`.

Comment (Yisaer) [reply pkg/manager/member/pd_member_manager.go:145]: If user fulfill the ClusterIP in the tidbcluster spec, shall we update the Service ClusterIP as user provided?

Comment (DanielZhangQD) [reply pkg/manager/member/pd_member_manager.go]: ClusterIP cannot be updated after the initial set.
If users set ClusterIP during the cluster creation, the clusterIP will be set in the `CreateService` function.

**Comment by DanielZhangQD** [comment]
@weekface @Yisaer comments addressed, PTAL again.

**Pull request merged**